src/reconcile/rec_write.c

```diff
@@ -1803,13 +1804,19 @@ __rec_split_write_supd(
          */
         r->supd_memsize = 0;
         for (j = 0; i < r->supd_next; ++j, ++i) {
-            /* Account for the remaining update memory. */
-            if (r->supd[i].ins == NULL)
-                /* Note: ins is never NULL for column-store */
-                upd = page->modify->mod_row_update[WT_ROW_SLOT(page, r->supd[i].rip)];
-            else
-                upd = r->supd[i].ins->upd;
-            r->supd_memsize += __wt_update_list_memsize(upd);
+            /* Only aggregate the size of updates that are restored. */
+            if (r->supd[i].restore) {
+                ++r->supd_restore_count;
+                if (r->supd[i].ins == NULL)
+                    /* Note: ins is never NULL for column-store */
+                    upd = page->modify->mod_row_update[WT_ROW_SLOT(page, r->supd[i].rip)];
+                else
+                    upd = r->supd[i].ins->upd;
+                for (; upd != NULL && upd != r->supd[i].onpage_upd &&
+                     upd != r->supd[i].onpage_tombstone;
+                     upd = upd->next)
+                    r->supd_memsize += upd->size;
+            }
```

Comment:
I'm confused by this code. I think the code populating the list of saved updates, also populates the memory size. This code is trimming that list of saved updates down as the updates are distributed across different pages, but it's also re-incrementing the memory size for any updates that weren't removed.

Should it instead be decrementing the size of updates that have been associated with a different page? Or possibly not doing anything at all?

Reply:
I think what it does is to shift the saved updates that have been removed to the split page. In line 1805, it first reset the size to 0, so we should recalculate the size again.

Reply:
I see - should it also be resetting `supd_restore_count` to 0?

Reply:
Good point. Fixed in [d4aecbe](https://github.com/wiredtiger/wiredtiger/pull/8127/commits/d4aecbeddf100340aaaa98f022cc040aed90083f)